### PR TITLE
Added support for receiver ack to control cursor

### PR DESCRIPTION
**Pull request by taglines**
Fix selinix policy so output sockets can be placed in /var/run

### CODE REVIEW

File: Event.cpp [hunk 1]
```diff
@@ -195,39 +197,39 @@ constexpr uint32_t RECORD_HEADER_SIZE(uint16_t num_fields, uint16_t name_size, u
 
 constexpr uint32_t FIELD_NAME_SIZE_OFFSET = 0;
 constexpr uint32_t FIELD_NAME_SIZE_SIZE = sizeof(uint16_t);
-inline uint16_t& FIELD_NAME_SIZE(uint8_t* data, uint32_t field_offset) {
-    return *reinterpret_cast<uint16_t*>(data+field_offset);
+inline uint16_t& FIELD_NAME_SIZE(uint8_t* data, uint32_t record_offset, uint32_t field_offset) {
```
Comment (xi-chen) [inline Event.cpp]: In general, do we need to do bound check when we casting to avoid overflow.

File: IO.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    int fd = _fd.load();
+    if (_fd < 0 || _rclosed.load()) {
+        return CLOSED;
+    }
+    struct pollfd fds;
+    fds.fd = fd;
+    fds.events = POLLIN;
+    fds.revents = 0;
+
+    auto ret = poll(&fds, 1, static_cast<int>(timeout));
```
Comment (xi-chen) [inline IO.cpp]: I notice the timeout is -1, do we need a smaller timeout to make sure that poll will return and do not block other work?

Comment (taglines) [reply IO.cpp]: In cases where the timeout is -1, if the thread blocked on poll needs to be terminated, the thread is sent SIGQUIT and the poll will be interrupted and will return with errno == EINTR. See RunBase.cpp for the core thread/task management logic.

File: Event.cpp (hunk 1 above)
Comment (taglines) [reply Event.cpp]: Any overflow would be due to programmer error, not invalid/malicious input.

**Pull request merged**